apps/src/studio/studio.js

```diff
@@ -1586,7 +1627,15 @@ Studio.willCollidableTouchWall = function (collidable, xCenter, yCenter) {
     collisionRects = skin.customObstacleZones[Studio.background][Studio.wallMapRequested];
   }
 
-  if (collisionRects) {
+  var wallMapLayer = null;
+  if (Studio.wallMapLayers) {
+    wallMapLayer = Studio.wallMapLayers[Studio.wallMapRequested];
+  }
+
+  if (wallMapLayer) {
+    // Comapre against a layout image
+    return wallMapLayer[4 * (400 * yCenter + xCenter)] === 0;
```

Comment:
I don't understand what this line does, _and_ it has magic numbers (I assume the 400 is playspace size).